knockserver/views.py

```diff
@@ -6,8 +6,90 @@
 
 from knockserver import app
 from knockserver.database import db_session
-from knockserver.models import ExampleObject
+from knockserver.models import AccessPattern, PatternPiece
 
 @app.route('/')
 def index():
     return render_template('index.html')
+
+@app.route('/patterns/', methods=['POST'])
+def patterns_post():
+    """
+    Handles the creation of new knock patterns for a user.
+    """
+    data = request.get_json(force=True)# converts request body json into python dict
+
+    pattern = AccessPattern()
+    
+    # Fields from form
+    try:
+        pattern.name = data['name']
+    except KeyError as e:
+        return Response('Name is required', 400)
+
+    if 'expiration' in data:
+        pattern.expiration = data['expiration']
+    else:
+        pattern.expiration = 0
+
+    if 'maxUses' in data:
+        pattern.maxUses = data['maxUses']
+    else:
+        pattern.maxUses = -1
+
+    # Fields that always are initialized to the same value
+    pattern.active = True
+    pattern.usedCount = 0
+    pattern.pending = True
+
+    db_session.add(pattern)
+    db_session.commit()
+
+    return '{"success": true}' #TODO decide response
+
+
+@app.route('/patterns/', methods = ['GET'])
+def patterns_get():
+    """
+    Return all patterns for debugging
+    Patterns ordered newest first
+    """
+    return str(AccessPattern.query.order_by(AccessPattern.id.desc().all()))
```

Comment:
Also, the parentheses are incorrect on this. `.all()` should be called on `query`.
I'm about to push a fix for that.